proxy-node-gateway/src/test/java/uk/gov/ida/notification/apprule/rules/TestErrorPageExceptionMapperRule.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.ida.notification.apprule.rules;
+
+import io.dropwizard.testing.junit.ResourceTestRule;
+import org.glassfish.jersey.client.ClientProperties;
+
+import org.junit.ClassRule;
+import org.junit.Test;
+import uk.gov.ida.notification.exceptions.mappers.ErrorPageExceptionMapper;
+import uk.gov.ida.notification.exceptions.mappers.GenericExceptionMapper;
+
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestErrorPageExceptionMapperRule {
+
+    private static final String EIDAS_ERROR_PAGE_URI = "http://eidas_error_page_uri";
+    private static final String GENERIC_ERROR_PAGE_URI = "http://generic_error_page_uri";
+
+    @ClassRule
+    public static final ResourceTestRule resources = ResourceTestRule.builder()
+                                    .addProvider(new ErrorPageExceptionMapper(getUri(EIDAS_ERROR_PAGE_URI)))
+                                    .addProvider(new GenericExceptionMapper(getUri(GENERIC_ERROR_PAGE_URI)))
+                                    .addResource(new TestExceptionMapperResource())
+                                    .build();
+
+    @Test
+    public void shouldMapEidasSamlParserReponseExceptionToErrorPageExceptionMapper() {
+        Response response;
+
+        response = resources.getJerseyTest()
```

Comment:
This is not essential but all these tests can be slightly refactored to be even simpler.

Since the same client with the same set-up is used in every test, this can be refactored. Add a class field `testClient` and then a `setUp()` method with `@Before`  attribute to initialise it. `@Before` will make the method run before each test. Initialise the client with the common bit in each test (`RESOURCES.getJerseyTest()
                            .client().property(ClientProperties.FOLLOW_REDIRECTS, Boolean.FALSE))`
Then the `response` declaration and assignment can be combined to just `Response response = testClient.target(...)...`

Reply:
I've done this which is even simpler (and removes the mental gymnastics of @Before):

```java
    private Response getResponse(String uri) {
        return resources.getJerseyTest()
                        .client().property(ClientProperties.FOLLOW_REDIRECTS, Boolean.FALSE)
                        .target(uri).request().get();
    }
```

Reply:
Beautiful 👍